apps/health_campaign_field_worker_app/lib/blocs/search_households/search_households.dart

```diff
@@ -434,134 +360,103 @@ class SearchHouseholdsBloc
     // Perform a series of asynchronous data retrieval operations based on the search criteria.
 
     // Fetch household results based on proximity and other criteria.
-    final List<HouseholdModel> proximityBasedResults =
-        await addressRepository.searchHouseHoldbyAddress(AddressSearchModel(
-      latitude: event.latitude,
-      longitude: event.longitude,
-      maxRadius: event.maxRadius,
-    ));
 
-    // Fetch individual results based on proximity and other criteria.
-    final List<IndividualModel> proximityBasedIndividualResults =
-        await addressRepository.searchHouseHoldByIndividual(AddressSearchModel(
-      latitude: event.latitude,
-      longitude: event.longitude,
-      maxRadius: event.maxRadius,
-    ));
+    List<IndividualModel> indResults = [];
+    List<IndividualModel> proximityBasedIndividualResults = [];
 
+    if (event.isProximityEnabled) {
+      // Fetch individual results based on proximity and other criteria.
+      proximityBasedIndividualResults = await addressRepository
+          .searchHouseHoldByIndividual(AddressSearchModel(
+        latitude: event.latitude,
+        longitude: event.longitude,
+        maxRadius: event.maxRadius,
+      ));
+    }
     // Extract individual IDs from proximity-based individual results.
-    final indIds = proximityBasedIndividualResults
+    final List<String> indIds = proximityBasedIndividualResults
         .map((e) => e.clientReferenceId)
         .toList();
 
+    indResults = await individual.search(
+      event.isProximityEnabled
+          ? IndividualSearchModel(
+              clientReferenceId: indIds,
+              name: NameSearchModel(givenName: event.searchText.trim()),
+            )
+          : IndividualSearchModel(
+              name: NameSearchModel(givenName: event.searchText.trim()),
+            ),
+    );
+
+    final individualClientReferenceIds =
+        indResults.map((e) => e.clientReferenceId).toList();
     // Search for individual results using the extracted IDs and search text.
-    final List<IndividualModel> indResults = await individual.search(
-      IndividualSearchModel(
-        clientReferenceId: indIds,
-        name: NameSearchModel(givenName: event.searchText.trim()),
-      ),
+    final List<HouseholdMemberModel> householdMembers =
+        await fetchHouseholdMembersBulk(
+      individualClientReferenceIds,
+      null,
     );
 
-    // Search for individual results based on the search text only.
-    final List<IndividualModel> results = await individual.search(
-      IndividualSearchModel(
-        name: NameSearchModel(givenName: event.searchText.trim()),
+    final househHoldIds = beneficiaryType != BeneficiaryType.individual
+        ? householdMembers.map((e) => e.householdClientReferenceId!).toList()
+        : null;
```

Comment:
Why do we need to check for beneficiary type here, this will break for household type search i think please check once. @naveen-egov